pkg/api/auth_middleware.go

```diff
@@ -20,54 +23,88 @@ var (
 	ErrAuthenticationFailed    = errors.New("error authenticating request")
 )
 
-func AuthMiddleware(logger logging.Logger, authService auth.Service) func(next http.Handler) http.Handler {
+// extractSecurityRequirements using swagger return an array of security requirements set for the request
+func extractSecurityRequirements(router routers.Router, r *http.Request) (openapi3.SecurityRequirements, error) {
+	// Find route
+	route, _, err := router.FindRoute(r)
+	if err != nil {
+		return nil, err
+	}
+	if route.Operation.Security == nil {
+		return route.Swagger.Security, nil
+	}
+	return *route.Operation.Security, nil
+}
+
+func AuthMiddleware(logger logging.Logger, swagger *openapi3.Swagger, authService auth.Service) func(next http.Handler) http.Handler {
+	router, err := legacy.NewRouter(swagger)
+	if err != nil {
+		panic(err)
+	}
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			ctx := r.Context()
-			var user *model.User
-			var err error
-
-			// validate jwt token from cookie
-			jwtCookie, _ := r.Cookie(JWTCookieName)
-			if jwtCookie != nil {
-				user, err = userByToken(ctx, logger, authService, jwtCookie.Value)
-				if err != nil {
-					writeError(w, http.StatusUnauthorized, err)
-					return
-				}
+			securityRequirements, err := extractSecurityRequirements(router, r)
+			if err != nil {
+				writeError(w, http.StatusBadRequest, err)
```

Comment:
If this goes to users, is the return from `router.FindRoute` safe to show users (and friendly)?

Reply:
1. The possibilities: "no matching operation was found" or "method not allowed"
2. At this point this call should not fail - The first middleware that responsible to verify the request does the route lookup first. We have 3 places and this is the second one where we are required to do the same lookup again to get the relevant information related to this call.